File: bike-shop/backend/src/main/java/com/example/bike_shop/service/ReviewService.java
package com.example.bike_shop.service;
 
import com.example.bike_shop.dto.ReviewDTO;
 
import java.util.List;
 
public interface ReviewService {
    List<ReviewDTO> getByProduct(Long productId);
    ReviewDTO create(Long userId, Long productId, ReviewDTO dto);
    void delete(Long userId, Long reviewId, boolean isAdmin);
    boolean canReview(Long userId, Long productId);
}

File: bike-shop/frontend/src/services/reviewService.js
import axiosClient from "./axiosConfig";

const reviewService = {
  getByProduct: (productId) => axiosClient.get(`/reviews/product/${productId}`),
  canReview: (productId) => axiosClient.get(`/reviews/product/${productId}/can-review`),
  create: (productId, data) => axiosClient.post(`/reviews/product/${productId}`, data),
  delete: (reviewId) => axiosClient.delete(`/reviews/${reviewId}`),
};

export default reviewService;
